### PR TITLE
nats: add typing to main connect function

### PR DESCRIPTION
**Pull request by squat**
Currently, the `connect` functions of the main NATS modules are not
fully typed or typed slightly incorrectly. The examples all show that a
single string can be given for the `servers` parameter, but the `Client`
class does not declare that it accepts this type, which causes static
analysis issues when the module is used directly.

This commit extends the signatures for `connect` so that `servers` can
also be a `str`, effectively giving it the same type as what is expected
by `_setup_server_pool`.

Signed-off-by: Lucas Servén Marín <lserven@gmail.com>

### CODE REVIEW

File: nats/aio/client.py [hunk 1]
```diff
@@ -219,7 +219,7 @@ def __init__(self) -> None:
 
     async def connect(
         self,
-        servers: List[str] = ["nats://127.0.0.1:4222"],
+        servers: Union[str, List[str]] = ["nats://127.0.0.1:4222"],
```
Comment (squat) [inline nats/aio/client.py:222]: btw, what do you think about adding another commit so that both the `connect` methods use the same default, either `localhost` or `127.0.0.1`?

Comment (wallyqs) [reply nats/aio/client.py:222]: Thanks, maybe let's use `localhost` for both? wdyt?

**Review by wallyqs — APPROVED** [review]
LGTM

File: nats/aio/client.py (hunk 1 above)
Comment (squat) [reply nats/aio/client.py:222]: Agreed, this way we are not defaulting to IPv4. I'll add another commit to address that.

Comment (squat) [reply nats/aio/client.py]: ok ready :+1:

**Review by wallyqs — APPROVED** [review]
